MiniTwitch.Helix/Internal/Models/RequestData.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System.Text;
+
+namespace MiniTwitch.Helix.Internal.Models;
+
+internal class RequestData
+{
+    public object Body { get; set; } = default!;
+
+    internal readonly string _method;
+    private readonly string _url;
+    private readonly StringBuilder _paramBuilder = new();
+
+    public RequestData(string requestUrl, HelixEndpoint endpoint)
+    {
+        _url = requestUrl + endpoint.Route;
+        _method = endpoint.Method.Method;
+    }
+
+    public string GetUrl() => _url + _paramBuilder.ToString();
+
+    public RequestData AddParam(string key, object? value)
+    {
+        const char question = '?';
+        const char ampersand = '&';
+        if (value is null)
+            return this;
+
+        _ = _paramBuilder.Length == 0
```

Comment:
this looks like using language features in non-intended ways to get away from writing few additional lines of code